src/t_zset.c

```diff
@@ -72,12 +72,41 @@ void zsetConvertAndExpand(robj *zobj, int encoding, unsigned long cap);
 zskiplistNode *zslGetElementByRankFromNode(zskiplistNode *start_node, int start_level, unsigned long rank);
 zskiplistNode *zslGetElementByRank(zskiplist *zsl, unsigned long rank);
 
+static inline unsigned long zslGetNodeSpanAtHeight(zskiplistNode *x, int height) {
+    return height > 0 ? x->level[height].span : x->level[height].forward ? 1
+                                                                         : 0;
+}
+
+static inline void zslSetNodeSpanAtHeight(zskiplistNode *x, int height, unsigned long span) {
+    if (height > 0)
+        x->level[height].span = span;
+}
```

Comment:
So the span at loevel 0 can change and either be 1 or 0. In case (for example) a node is deleted from the end of the list the node preceeding it will be set to span 0.